tests/integration/reftests/test_pre_initial.py

```diff
@@ -102,8 +102,8 @@ async def test_over_bracketed(flow, scheduler, reftest):
             'final cycle point': '2013-12-25T12:00Z',
             'graph': {
                 'T12': '''
-                    (a[-P1D]:fail | b[-P1D]:fail | c[-P1D]:fail) => d
-                    a & b & c  # Implied by implicit cycling now...
+                    (a[-P1D]:fail? | b[-P1D]:fail? | c[-P1D]:fail?) => d
```

Comment:
This is absolutely reasobable in the case of recurring cycling, but is it reasonable in general?

i.e. I think that this should fail validation

```cylc
[graph]
    R1/1 = a
    R1/2 = a[-P1]:fail
```

but I'm not sure that _this_ should, because a at -P1 is not the same task as a.

```cylc
R1 = a[-P1]:fail => a
```

I think I may be happy to gloss over this, because the latter recurrance isn't very meaningful.

Reply:
So this test was passing on master only because the offsets meant we weren't checking those outputs for consistency with non-offset outputs. 

I agree that "a at -P1 is not the same task as a" but (for non-R1 cases at least) one becomes the other as the cycle point advances, so I think the output optionality does have to be consistent between the two. (If that makes sense!).

Reply:
Happy, unwilling to hold this PR up. As observed elsewhere this code is ripe for refactoring.